core/src/main/java/com/sap/ai/sdk/core/AiCoreDeployment.java

```diff
@@ -0,0 +1,224 @@
+package com.sap.ai.sdk.core;
+
+import com.sap.ai.sdk.core.client.DeploymentApi;
+import com.sap.ai.sdk.core.client.model.AiDeployment;
+import com.sap.ai.sdk.core.client.model.AiDeploymentList;
+import com.sap.cloud.sdk.cloudplatform.connectivity.DefaultHttpDestination;
+import com.sap.cloud.sdk.cloudplatform.connectivity.Destination;
+import com.sap.cloud.sdk.cloudplatform.connectivity.DestinationProperty;
+import com.sap.cloud.sdk.services.openapi.apiclient.ApiClient;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import javax.annotation.Nonnull;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+/** Connectivity convenience methods for AI Core with deployment. */
+@RequiredArgsConstructor(access = AccessLevel.PROTECTED)
+public class AiCoreDeployment implements AiCoreDestination {
+
+  private static Map<String, AiDeploymentList> CACHE = new LinkedHashMap<>();
+
+  private static final String AI_RESOURCE_GROUP = "URL.headers.AI-Resource-Group";
+
+  // the delegating AI Core Service instance
+  @Nonnull private final AiCoreService service;
+
+  // the deployment id handler to be used, based on instance
+  @Nonnull private final Function<AiCoreDeployment, String> deploymentId;
+
+  // the resource group, "default" if null
+  @Getter(AccessLevel.PROTECTED)
+  @Nonnull
+  private final String resourceGroup;
+
+  /**
+   * Default constructor with "default" resource group.
+   *
+   * @param service The AI Core service.
+   * @param deploymentId The deployment id handler.
+   */
+  public AiCoreDeployment(
```

Comment:
```suggestion
  protected AiCoreDeployment(
```

Reply:
I can't change it to non-public because the method `withResourceGroup(String)` will be inaccessible then. (Right?)

Reply:
Nope, still accessible from the `OpenAiClient` for example

Reply:
I did it in my follow up PR